From: Xiao Ni <xni@redhat.com>
Date: Sun, 12 Jun 2016 17:18:00 +0800
Subject: MD:Update superblock when err == 0 in size_store
References: bsc#1003941,FATE#321732
Git-commit: 4ba1e78891e9858a5c773e6ffe4414611019bf42
Patch-Mainline: v4.8-rc1

This is a simple check before updating the superblock. It should update
the superblock when update_size return 0.

Signed-off-by: Xiao Ni <xni@redhat.com>
Signed-off-by: Shaohua Li <shli@fb.com>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/md/md.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 5267f1e..5b8cc8c 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -4261,7 +4261,8 @@ size_store(struct mddev *mddev, const char *buf, size_t len)
 		return err;
 	if (mddev->pers) {
 		err = update_size(mddev, sectors);
-		md_update_sb(mddev, 1);
+		if (err == 0)
+			md_update_sb(mddev, 1);
 	} else {
 		if (mddev->dev_sectors == 0 ||
 		    mddev->dev_sectors > sectors)
-- 
1.8.5.6

